### PR TITLE
Starting 2.0

### PR DESCRIPTION
**Pull request by ffissore**
This PR sets the minimum java version to 1.8 and includes slf4j-fluent

### CODE REVIEW

**Comment by ffissore** [comment]
Uhm build failed in module slf4j-jdk14 but I cannot reproduce

**Comment by ceki** [comment]
FYI, it is better to rebase rather than pull/merge.

**Comment by ceki** [comment]
Instead of pulling in a new dependency, it better more sense to enhance the existing API, more specifically the logger interface

**Comment by ffissore** [comment]
Ok, moved the code into `slf4j-api` module. I've kept module `slf4j-fluent` (renamed to `-test`) because tests depend on logback, and having them in `slf4j-api` would have created a circular dependency.

WRT your suggestion to rebase, I usually do. If you were referring to your commits mixed with mine that is because I based my work on `master` but I'm targeting `2.0`, which is older

**Comment by ffissore** [comment]
For the record, the failed build is again caused by `JDK14MultithreadedInitializationTest`. Before I force-pushed a fix, the build was green, and then it turned red again, so it seems quite erratic

File: slf4j-api/src/main/java/org/slf4j/Logger.java [hunk 1]
```diff
@@ -63,6 +70,23 @@
...
+     *
+     * ...
+     *
+     * logger.atError().log("This is a {}", lazy(() -> "test"));
+     * </pre>
+     *
+     * @param supplier the lambda to cast
+     * @return a Supplier
+     */
+    static Supplier<?> lazy(Supplier<?> supplier) {
```
Comment (jliu666) [inline slf4j-api/src/main/java/org/slf4j/Logger.java]: Maybe 
```java
logger.atError().log("This is a {}", () -> "test");
```
is easier to use than 
```
logger.atError().log("This is a {}", lazy(() -> "test"));
```

I think both should be supported.
Why not?

Comment (ffissore) [reply slf4j-api/src/main/java/org/slf4j/Logger.java]: It does not compile

Comment (jliu666) [reply slf4j-api/src/main/java/org/slf4j/Logger.java]: When will this version release? slf4j community doesn't seem to be active ?

Comment (jliu666) [inline slf4j-api/src/main/java/org/slf4j/Logger.java]: add a function to logger:
```java
public void info(String string, Supplier<String> msg);

this compile.
logger.info("This is a {}", () -> "test");
```
More functions in logger makes user less code.

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
 language: java
+
 jdk:
-  - oraclejdk9
+  - oraclejdk11
+
 notifications:
   email:
-      - notification@qos.ch
+      - notification@qos.c
```
Comment (twillouer) [inline .travis.yml]: strange, not ?

Comment (ffissore) [reply .travis.yml]: Good catch. That's 5ddfa9c6b3252834cf3be5f88ea27c821f70a577 which is now on master. @ceki you want to fix that

Comment (ceki) [reply .travis.yml]: @twillouer  thank you. Fixed in the 2.0 branch

**Pull request closed without merging**